modules/data_release/ajax/AddPermission.php

```diff
@@ -20,9 +20,16 @@
 $baseURL  = $settings->getBaseURL();
 $DB       = $factory->database();
 
-if (isset($_POST['action']) && $_POST['action'] == 'addpermission'
+if (isset($_GET['action']) && $_GET['action'] == 'addpermission'
```

Comment:
Do you think you could rewrite this in the "happy path" format?

Basically it should fail early if `'action'` is not set and from there proceed knowing that the value is good (maybe with a switch-case instead of if-else).

Reply:
should be done in the next commit.